### PR TITLE
Fix writing of nodes.conf

### LINKED ISSUE #779: write operations to nodes.conf no longer work
### Version of Warewulf

```console
$ sudo wwctl version
wwctl version:	 4.4.0-1.git_0f2fcb3
rpc version: apiPrefix:"rc1"  apiVersion:"1"  warewulfVersion:"4.4.0-1.git_0f2fcb3"
```

### Expected behavior

Certain commands, such as `wwctl node set`, `wwctl profile set`, `wwctl node edit`, and `wwctl profile edit` should change the contents of `nodes.conf`.

### Actual behavior

These commands do not actually change the contents of `nodes.conf`.

### Steps to reproduce this behavior

```console
$ sudo wwctl node list n1 -a | grep hwaddr
  n1    default:hwaddr      --          de:ad:be:ef:00:00                                     

$ sudo wwctl node set n1 --hwaddr DE:AD:BE:EF:DE:AD
Are you sure you want to modify 1 nodes(s): y

$ sudo wwctl node list n1 -a | grep hwaddr
  n1    default:hwaddr      --          de:ad:be:ef:00:00
```

### What OS/distro are you running

```
NAME="Rocky Linux"
VERSION="9.1 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.1"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.1 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.1"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.1"
```

### How did you install Warewulf

RPMs built locally from git.

### PR DESCRIPTION
**Pull request by anderbubble**
The caching functionality, and the optional persistence functionality that came with it, appears to have been preventing actually writing the configuration to disk in all cases. This change fixes the ability to write to nodes.conf.

Somewhat opinionatedly, this change removes the caching functionality of the nodes.conf and warewulf.conf configuration. The tests have also been updated to use real files for i/o rather than tying into the caching system, making them more accurately test the ability of Warewulf to actually read and write its configuration.

Fixes #779

### CODE REVIEW

**Comment by anderbubble** [comment]
This PR breaks reading warewulf.conf, so I'm switching it to a draft while I fix that.

**Comment by mslacken** [comment]
The caching functionality is part of warewulf since the start and I just touched it for `warewulf.conf`  when adding the test functionality. When doing this I also added the caching for the `nodes.conf`.
I am completely with you, by dropping this caching as I do not see an obvious big performance issue here, but we ask @gmkurtzer about this, as he implemented the caching in the first place.

**Comment by anderbubble** [comment]
I decided that my changes to warewulfconf were over-zealous for this PR, and weren't necessary to fix the specific issue, so I've backed them out leaving only the nodes changes.

File: internal/pkg/node/constructors.go [hunk 1]
```diff
@@ -47,39 +45,43 @@ func init() {
...
-
 	wwlog.Verbose("Opening node configuration file: %s", ConfigFile)
 	data, err := os.ReadFile(ConfigFile)
 	if err != nil {
-		return ret, err
+		return NodeYaml{}, err
 	}
+	return Parse(data)
+}
+
```
Comment (mslacken) [inline internal/pkg/node/constructors.go:62]: Please add a comment what exactly Parse is doing and what kind of errors can occure

**Review by mslacken — APPROVED** [review]

**Comment by anderbubble** [comment]
@mslacken requested comments added.

**Pull request merged**